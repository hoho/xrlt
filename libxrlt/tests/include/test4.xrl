<?xml version="1.0"?>
<xrl:requestsheet xmlns:xrl="http://xrlt.net/Transform">
    <xrl:response>
        <xrl:include>
            <xrl:href>/test/href/1</xrl:href>
            <xrl:type>json</xrl:type>
            <xrl:success select="/*" />
            <xrl:failure>
                <a>aaaa</a>
            </xrl:failure>
        </xrl:include>

        <xrl:include>
            <xrl:href>/test/href/2</xrl:href>
            <xrl:type>text</xrl:type>
            <xrl:success select="substring-before(/, '123')" />
            <xrl:failure>
                <b>bbbb</b>
            </xrl:failure>
        </xrl:include>

        <xrl:include>
            <xrl:href select="'/test/href/3'" />
            <xrl:type>xml</xrl:type>
            <xrl:failure>
                <c>cccc</c>
            </xrl:failure>
        </xrl:include>
    </xrl:response>
</xrl:requestsheet>
