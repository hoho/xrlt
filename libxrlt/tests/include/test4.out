XRLT_STATUS_SUBREQUEST
sr id: 1
sr method: GET
sr type: JSON
sr url: /test/href/1
sr query: (null)
sr body: (null)
XRLT_STATUS_REFUSE_SUBREQUEST
XRLT_STATUS_SUBREQUEST
sr id: 2
sr method: GET
sr type: TEXT
sr url: /test/href/2
sr query: (null)
sr body: (null)
XRLT_STATUS_SUBREQUEST
sr id: 3
sr method: GET
sr type: XML
sr url: /test/href/3
sr query: (null)
sr body: (null)
XRLT_STATUS_CHUNK
chunk: aaaa
XRLT_STATUS_WAITING
XRLT_STATUS_REFUSE_SUBREQUEST
XRLT_STATUS_CHUNK
chunk: bbbb
XRLT_STATUS_WAITING
XRLT_STATUS_REFUSE_SUBREQUEST
XRLT_STATUS_CHUNK
chunk: cccc
XRLT_STATUS_DONE
